cfpb/sbl-frontend
# [Refactor] Filing navigation buttons

Close #448
 
~~**NOTE**: This branch builds on top of the changes from the following tickets, so will need to wait for those to all be merged into main in order to avoid major merge conflicts:
#455 
#291 
#450 
#476~~

## Changes

- [FilingNavButtons] Create a consolidated component capable of handling the submit/loading/go-back/clear-form functionality of all current pages
- Update all pages to use the new component
- Some small language and loading state fixes

## How to test this PR

1. Run through the whole flow (with an Institution that does not initially have Types) and verify that the buttons look good

## Note
- Screenshots do not reflect the updated label from `Go back to previous step` to now being `Go back`, but this is implemented.
- Screenshots don't reflect it but I've corrected the `Clear form` button that was showing as blue but now shows as red


## Screenshots

1. Type of FI shows loading state

https://github.com/cfpb/sbl-frontend/assets/2592907/6c099ca9-552f-4676-b121-54ee42289b2f

2. Upload blocked until upload is complete

|Upload incomplete|Upload complete|
|---|---|
|<img width="793" alt="Screenshot 2024-05-14 at 11 48 23 AM" src="https://github.com/cfpb/sbl-frontend/assets/2592907/39546219-ee2a-4560-ad67-7c8a0f8d9266">|<img width="797" alt="Screenshot 2024-05-14 at 11 47 52 AM" src="https://github.com/cfpb/sbl-frontend/assets/2592907/b02d4243-8883-4f6b-9c72-c74bb1f76a85">|



3. Errors complete

|Part 1 |Part2|
|---|---|
|<img width="788" alt="3-errors__1-none" src="https://github.com/cfpb/sbl-frontend/assets/2592907/143b1b35-9da0-4270-bddb-51f51c75da8b">|<img width="793" alt="3-errors__2-none" src="https://github.com/cfpb/sbl-frontend/assets/2592907/e23b7252-329a-4d68-b459-c87cc9ace440">|
4. Warnings show loading state

https://github.com/cfpb/sbl-frontend/assets/2592907/fb998c2d-d7c8-4710-9add-6d0a42524f07



5. Point of Contact shows loading state

https://github.com/cfpb/sbl-frontend/assets/2592907/2ecb5f00-3cd7-46f1-9b9c-1756afcd423c



6. Sign and submit does not allow further progress
<img width="561" alt="6-sign-submit__done" src="https://github.com/cfpb/sbl-frontend/assets/2592907/51005b18-118d-49b2-9e8c-204a516a952f">
